suite-native/app/src/navigation/RootStackNavigator.tsx

```diff
@@ -35,40 +43,51 @@ export const RootStackNavigator = () => {
         return RootStackRoutes.Onboarding;
     };
 
+    // Take screenshot on navigation state change so it can be used for biometrics and other overlays later.
+    useEffect(() => {
+        const removeStateSubscription = navigation.addListener('state', () => {
+            takeScreenshot(screenshotViewRef);
+        });
+
+        return removeStateSubscription;
+    }, [navigation, takeScreenshot]);
+
     return (
-        <RootStack.Navigator
-            initialRouteName={getInitialRouteName()}
-            screenOptions={stackNavigationOptionsConfig}
-        >
-            <RootStack.Screen
-                name={RootStackRoutes.Onboarding}
-                component={OnboardingStackNavigator}
-            />
-            <RootStack.Screen name={RootStackRoutes.AppTabs} component={AppTabNavigator} />
-            <RootStack.Screen
-                name={RootStackRoutes.AccountsImport}
-                component={AccountsImportStackNavigator}
-            />
-            <RootStack.Screen
-                options={{ title: RootStackRoutes.AccountSettings }}
-                name={RootStackRoutes.AccountSettings}
-                component={AccountSettingsScreen}
-            />
-            <RootStack.Screen
-                options={{ title: RootStackRoutes.TransactionDetail }}
-                name={RootStackRoutes.TransactionDetail}
-                component={TransactionDetailScreen}
-            />
-            <RootStack.Screen
-                options={{ title: RootStackRoutes.AccountDetail }}
-                name={RootStackRoutes.AccountDetail}
-                component={AccountDetailScreen}
-            />
-            <RootStack.Screen
-                name={RootStackRoutes.DevUtilsStack}
-                component={DevUtilsStackNavigator}
-            />
-            <RootStack.Screen name={RootStackRoutes.ReceiveModal} component={ReceiveModal} />
-        </RootStack.Navigator>
+        <View style={StyleSheet.absoluteFillObject} ref={screenshotViewRef}>
```

Comment:
Why is this not in the screenshot provider?

Reply:
Originally I had this wrapper view in the `ScreenshotProvider` and I agree that it would be the nicer solution. But unfortunately, in that case, the app crashed on every call of the `takeScreenshot` function.

I am afraid that the Skia is not able to handle a screenshot of a view that contains modals. Since the children of the `ScreenshotProvider` view container the RooStackNavigator but also modals such as `AlertRender`, `MessageSystemRenderer` etc. we cannot put the screenshot view in there.

Reply:
after discussion the screenshot capturing logic moved to separate `ScreenshotCapturer` component.